distributed/shuffle/tests/test_multi_comm.py

```diff
@@ -0,0 +1,22 @@
+from collections import defaultdict
+
+import pytest
+
+from distributed.shuffle.multi_comm import MultiComm
+
+
+@pytest.mark.asyncio
+async def test_basic(tmp_path):
+    d = defaultdict(list)
+
+    async def send(address, shards):
+        d[address].extend(shards)
```

Comment:
What happens if send raises an `OSError`? What happens if it raises _any_ error?